src/blk/aio/aio.cc

```diff
@@ -25,33 +25,43 @@ int aio_queue_t::submit_batch(aio_iter begin, aio_iter end,
   int r;
 
   aio_iter cur = begin;
-  struct aio_t *piocb[aios_size];
```

Comment:
The problem, as I understand it, is very simple: `aios_size` got an overflow.
I think solution should be to either:
1) change `uint16_t` to `uint32_t` (preferred)
2) count elements between begin and end; use `alloca` to allocate piocb.

The only benefit of current solution is that it does minimize stack allocation, but at cost of added complexity. I do not think its worth it.
Note that for every entry in `piocb` we will be reading at least 4K from the drive. I think we can afford having entire `piocb` on stack.

Reply:
@aclamk - you might be missing one more aspect - amount of chunks (and hence amount IOs issued in a single batch) in an object is determined (to some degree)  by cluster's users. So it becomes possible for them to use "bad" objects which could DDOS OSDs or do something inappropriate. 
And this patch provides a sort of protection against that - at aio level at least..